src/snowflake/snowpark/mock/_plan.py

```diff
@@ -535,6 +543,25 @@ def cleanup_imports():
             function_input[col_name] = calculate_expression(
                 child, input_data, analyzer, expr_to_alias
             )
+        # Validate function inputs
+        actual_types = [col.datatype for col in function_input.sf_types.values()]
+        if len(actual_types) != len(udf._input_types):
+            raise SnowparkLocalTestingException(
+                f"Expected {len(udf._input_types)} arguments, but received {len(actual_types)}"
+            )
+
+        for expected, actual in zip(udf._input_types, actual_types):
+            if not types_are_compatible(expected, actual):
+                raise SnowparkLocalTestingException(
+                    f"UDF received input types {actual_types}, but expected inputs types of {udf._input_types}"
+                )
+
+        try:
+            res = function_input.apply(lambda row: udf_handler(*row), axis=1)
+        except Exception as err:
+            SnowparkLocalTestingException.raise_from_error(
+                err, error_message=f"Python Interpreter Error: {err}"
+            )
```

Comment:
I have changed the implementation in my PR: https://github.com/snowflakedb/snowpark-python/pull/1645/files

looks like GH treated this as new code instead of conflict, we need to merge them

Reply:
updated